fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -763,6 +763,23 @@ private String mintNewPid(final String slug) {
         return pid;
     }
 
+    /**
+     * perform save with checking to prevent SN-Sibling nodes from creating
+     * @param resource the FedoraResource need to save
+     * @throws RepositoryException
+     */
+    private void save(final FedoraResource resource) throws RepositoryException {
+        synchronized (LOGGER) {
```

Comment:
Yes, it look like something need to be done for concurrent support, which should be another story. The bug reported in [FCREPO-2055|https://jira.duraspace.org/browse/FCREPO-2055] is very specific case that relates to large file ingest only. I believe Modeshape doesn't handle well for nt:binary, which shouldn't allow SN-Sibling, so we are only trying to walk around it here ( see https://jira.duraspace.org/browse/FCREPO-2055?focusedCommentId=50506&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-50506 ).